javascripts/2048.js

```diff
@@ -1,32 +1,406 @@
+// from: http://stackoverflow.com/questions/7837456/comparing-two-arrays-in-javascript
+Array.prototype.equals = function (array) {
```

Comment:
Since the comparison is recursive this method could be called `deepEqual` for clarity.